health/service/src/test/scala/com/cerner/beadledom/health/resource/BaseSpec.scala

```diff
@@ -0,0 +1,57 @@
+package com.cerner.beadledom.health.resource
+
+import java.net.URI
+import java.util.Optional
+
+import com.cerner.beadledom.health.api.DependenciesResource
+import com.cerner.beadledom.health.{HealthDependency, HealthStatus}
+import com.cerner.beadledom.testing.Primitives
+import com.github.mustachejava.DefaultMustacheFactory
+import javax.ws.rs.core.{UriBuilder, UriInfo}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{FunSpec, MustMatchers}
+
+/**
+  * Base Spec for resource tests that provides a create dependency method
+  * for creating Health Dependencies.
+  *
+  * @author Nick Behrens
+  */
+class BaseSpec extends FunSpec with MustMatchers with MockitoSugar {
+
+  protected val mockUriInfo: UriInfo = mock[UriInfo]
+  private val mockUriBuilder = mock[UriBuilder]
+  private val url = URI.create("demoUrl")
+
+  when(mockUriInfo.getBaseUriBuilder).thenReturn(mockUriBuilder)
+  when(mockUriBuilder.path(any[Class[DependenciesResource]])).thenReturn(mockUriBuilder)
+  when(mockUriBuilder.path(any[String])).thenReturn(mockUriBuilder)
+  when(mockUriBuilder.build()).thenReturn(url)
+
+  val mustacheFactory = new DefaultMustacheFactory("com/cerner/beadledom/health")
+
+  protected var primaryHealthyDependency: HealthDependency = createDependency(Primitives.generateString(), healthy = true, primary = true)
+  protected var ancillaryHealthyDependency: HealthDependency = createDependency(Primitives.generateString(), healthy = true, primary = false)
+  protected var primaryUnhealthyDependency: HealthDependency = createDependency(Primitives.generateString(), healthy = false, primary = false)
+  protected var ancillaryUnhealthyDependency: HealthDependency = createDependency(Primitives.generateString(), healthy = false, primary = false)
+
+
+  def createDependency(name: String, healthy: java.lang.Boolean, primary: Boolean): HealthDependency =
```

Comment:
`healthy` can also be a Scala Boolean

Reply:
Changed to a scala boolean: https://github.com/nab0310/beadledom/commit/4928608b1f95af3516d4c682c8c88b17c0832d15